### DISCUSSION TITLE
imgcat in new tab

### DISCUSSION THREAD

**Discussion by kenderipa** [wezterm/wezterm#4496] [Q&A] [answered]
### What Operating System(s) are you running on?

Linux X11

### Which Wayland compositor or X11 Window manager(s) are you using?

JWM

### WezTerm version

wezterm 20231004-113117-11dec45f

### Ask your question!

Hello,

I'm trying to open a picture in new tab with built-in tools, something like wezterm cli spawn -- wezterm imgcat picture name. It doesn't work. Am I missing something? Or is there another way to do that without external image viewers?

Thanks!

**Comment by wez** [comment, marked as answer]
`wezterm imgcat` terminates as soon as it is done outputting the image, so the associated pty will be closed.
You can use the `--hold` parameter to wait for enter to be pressed.

```console
$ wezterm cli spawn -- wezterm imgcat --hold /path/to/picture
```

**Reply by kenderipa** [reply]
Thanks! Is there a way to do it with bash? Like wezterm cli spawn -- bash -c "${wezterm imgcat /path/to/picture}"? I'm thinking about writing a simple image viewer in bash.

**Reply by megalithic** [reply]
You can try: `wezterm cli split-pane --right --percent 30 -- bash -c 'wezterm imgcat /path/to/pic ; read'`

This is what I presently use within neovim to open image files to a split in wezterm, then just hit `enter` to close it.

EDIT: just confirmed @wez solution works just fine, new version:

`wezterm cli split-pane --right --percent 30 -- bash -c 'wezterm imgcat --hold /path/to/pic'`
